### PR TITLE
VIP-10: Add node staking requirements

### PR DESCRIPTION
**Pull request by tahpot**
Add requirements for node staking and payouts

### CODE REVIEW

File: VIPs/vip-10.md [hunk 1]
```diff
@@ -31,40 +31,35 @@ The current storage node database is centralized and needs to be stored on-chain
...
-TBA
+- `STAKE_PER_SLOT` (const): The number of tokens that must be staked by a storage node operator for each storage slot they make available to the network. `default=60`. Only contract owner can change at this stage. _In the future this will become dynamic based on market dynamics._
 
 ## addNode(nodeInfo: StorageNode, requestSignature: string, requestProof: string, authSignature: string)
 
 Storage nodes must register themselves with the protocol so they can be discovered by new users. Storage nodes provide additional metadata (via `/status` endpoint) that assists users select the most appropriate storage nodes to use.
 
 It's not possible to re-use a DID to register multiple storage nodes. This is because storage endpoints sign their responses using the DID private key to enhance security of the protocol.
 
+The requester must provide VDA tokens to stake. The number of staked tokens must be `numberSlots` * `STAKE_PER_SLOT`. The staked tokens will be registered against the storage node being registered so they can be unstaked in the future (see `withdraw()`)
```
Comment (ITStar10) [inline VIPs/vip-10.md]: The `withdraw()` function can unstake only excess token when the `STAKE_PER_SLOT` is updated.
Staked tokens will be released by `removeNodeComplete()` function.

File: VIPs/vip-10.md
Comment (ITStar10) [inline VIPs/vip-10.md:1]: Please check the `addNode()` function. Added a comment.

**Review by ITStar10 — CHANGES_REQUESTED** [review]

File: VIPs/vip-10.md (hunk 1 above)
Comment (tahpot) [reply VIPs/vip-10.md]: I thought the comments in `withdraw()` and `removeNodeComplete()` made this clear?

Comment (ITStar10) [reply VIPs/vip-10.md]: How about to update the comments in the parenthesis as following?
```
(see `withdraw()`)
```
to
```
(see `withdraw` and `removeNodeComplete`)
```

Comment (tahpot) [reply VIPs/vip-10.md]: Got it. Good idea.

**Pull request merged**